server/src/main/java/org/opensearch/index/translog/RemoteFsTranslog.java

```diff
@@ -341,6 +345,13 @@ public void trimUnreferencedReaders() throws IOException {
         // clean up local translog files and updates readers
         super.trimUnreferencedReaders();
 
+        // Since remote generation deletion is async, this ensures that only one generation deletion happens at a time.
+        // Remote generations involves 2 async operations - 1) Delete translog generation files 2) Delete metadata files
+        // We try to acquire 2 permits and if we can not, we return from here itself.
+        if (remoteGenerationDeletionPermits.tryAcquire(2) == false) {
```

Comment:
Lets move 2 to constants

Reply:
Also trying to understand can 1 instance of segment deleter and 1 instance of translog deleter can run independently. What happens if segment deleter slows down translog deleter.

Reply:
@Bukhtawar  these are deletion of translog files and metadata files. Segments is not present in this scope. Also, since both the deletion uses the common generations to delete - https://github.com/opensearch-project/OpenSearch/pull/6400#discussion_r1112810046.

Reply:
Thanks for the details @ashking94 I am not sure but I think this breaks recovery semantic f.e. if the metadata is present we expect referenced translog files to be present as well but with this change will no longer continue to guarantee us that property. I propose we re-think about our GC strategy and see if we could do soft-deletes. It not only provides safety against deletions but also helps the future work around PITR

Reply:
In the light of the current PR lets move to constants and rename permits better and lets follow this through

Reply:
Created issue https://github.com/opensearch-project/OpenSearch/issues/6409. We will be revisit the remote deletion strategy considering PITR and other factors as mentioned in the mentioned issue.

Incorporated other comments as suggested.